src/main/kotlin/org/wycliffeassociates/otter/jvm/app/ui/takemanagement/view/RecordableFragment.kt

```diff
@@ -0,0 +1,215 @@
+            runOnNode?.let { it() }
+        }
+
+        fun selectedTakeContainer(runOnPlaceHolder: Node.() -> Unit) = VBox().apply {
+            alignment = Pos.CENTER
+
+            val placeholder = vbox {
+                runOnPlaceHolder()
+            }
+
+            recordableViewModel.selectedTakeProperty.onChangeAndDoNow {
+                clear()
+                when (it) {
+                    null -> add(placeholder)
+                    /* We can't just add the node being dragged, since the selected take might have just been
+                        loaded from the database */
+                    else -> add(createTakeCard(it))
+                }
+            }
+        }
+    }
+
+    protected val dragComponents = DragComponents()
+    private val dragTargetTop = dragComponents.dragTargetTop()
+
+    private val dragContainer = VBox().apply {
+        draggingNodeProperty.onChange { draggingNode ->
+            clear()
+            draggingNode?.let { add(draggingNode) }
+        }
+    }
+
+    init {
+        importStylesheet<AppStyles>()
+        createAudioPluginProgressDialog()
+    }
+
+    final override val root: Parent = anchorpane {
+        addDragTakeEventHandlers()
+        addButtonEventHandlers()
+
+        createSnackBar(this)
+
+        add(mainContainer.apply {
+            anchorpaneConstraints {
+                leftAnchor = 0.0
+                rightAnchor = 0.0
+                bottomAnchor = 0.0
+                topAnchor = 0.0
+            }
+        })
+        add(dragContainer)
+    }
+
+    private fun Parent.addDragTakeEventHandlers() {
+        addEventHandler(StartDragEvent.START_DRAG, ::startDrag)
+        addEventHandler(AnimateDragEvent.ANIMATE_DRAG, ::animateDrag)
+        addEventHandler(CompleteDragEvent.COMPLETE_DRAG, ::completeDrag)
+    }
+
+    private fun Parent.addButtonEventHandlers() {
+        addEventHandler(PlayOrPauseEvent.PLAY) {
+            lastPlayOrPauseEvent.set(it)
+        }
+        addEventHandler(DeleteTakeEvent.DELETE_TAKE) {
+            recordableViewModel.deleteTake(it.take)
+        }
+        addEventHandler(EditTakeEvent.EDIT_TAKE) {
+            recordableViewModel.editTake(it)
+        }
+    }
+
+    private fun createSnackBar(pane: Pane) {
+        // TODO: This doesn't actually handle anything correctly. Need to know whether the user
+        // TODO... hasn't selected an editor or recorder and respond appropriately.
+        val snackBar = JFXSnackbar(pane)
+        recordableViewModel.snackBarObservable.subscribe {
+            snackBar.enqueue(
+                JFXSnackbar.SnackbarEvent(
+                    messages["noRecorder"],
+                    messages["addPlugin"].toUpperCase(),
+                    5000,
+                    false,
+                    EventHandler {
+                        audioPluginViewModel.addPlugin(true, false)
+                    })
```

Comment:
})

Reply:
All of these have been fixed. I also just reread parts of the Kotlin Coding Conventions to help with this, since I was unclear about proper formatting for chained call wrapping.